### ISSUE TITLE
Sampling from z

### ISSUE THREAD

**Issue by extragoya** [RayDeeA/ibinn_imagenet#6]
I noticed that the sampling function seems to be out of date:
https://github.com/RayDeeA/ibinn_imagenet/blob/04b2ab9315e591506d6805e96c35b60222447071/ibinn_imagenet/model/classifiers/invertible_imagenet_classifier.py#L171

However, I'm having trouble doing a proper reverse sampling - for instance if use the following code snippet, I am unable to recreate the input from the output:

```
z_fc, z_conv = self.model(x)
with torch.no_grad():
    rev_ims = self.model([z_fc, z_conv], rev=True)
```

`x` will be in the range [0,1], but `rev_ims` will be in the range of roughly [-100, 100]. Is there something I'm doing trivially wrong here?

**Comment by extragoya** [comment]
I figured out the issue: I believe it's due to different statistics used in the BN of the subnetworks. This can cause very different activation maps. E.g., if you do forward in training mode, but sample backward in evaluation mode, the recreated input can be orders of magnitude different in scale. Likewise if you do forward with a full batch in training model but sample backward with a sub-batch. 

Have you tried using something like InstanceNorm instead, given the impact of 'non-invertibility' if the BN is used  in eval/training mode or with differently sized batches in training mode?

Anyway, I leave this issue open for now, in case my diagnosis was incorrect. Feel free to close. Any insights in the impact of BN would also be welcome - thanks!
